plugins/inputs/xtremio/xtremio.go

```diff
@@ -0,0 +1,388 @@
+package xtremio
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/tidwall/gjson"
+)
+
+type XtremIO struct {
+	Username   string   `toml:"username"`
+	Password   string   `toml:"password"`
+	URL        string   `toml:"url"`
+	Collectors []string `toml:"collectors"`
+	Cookie     *http.Cookie
+	tls.ClientConfig
+	Log telegraf.Logger
+}
+
+const sampleConfig = `
+## XtremIO User Interface Endpoint
+url = "https://xtremio.example.com/" # required
+
+## Credentials
+username = "user1"
+password = "pass123"
+
+## Metrics to collect from the XtremIO
+# collectors = ["bbus","clusters","ssds","volumes","xms"]
+
+## Optional TLS Config
+# tls_ca = "/etc/telegraf/ca.pem"
+# tls_cert = "/etc/telegraf/cert.pem"
+# tls_key = "/etc/telegraf/key.pem"
+## Use SSL but skip chain & host verification
+# insecure_skip_verify = false
+`
+
+// Description will appear directly above the plugin definition in the config file
+func (xio *XtremIO) Description() string {
+	return `Gathers Metrics From a Dell EMC XtremIO Storage Array's V3 API`
+}
+
+// SampleConfig will populate the sample configuration portion of the plugin's configuration
+func (xio *XtremIO) SampleConfig() string {
+	return sampleConfig
+}
+
+func (xio *XtremIO) Init() error {
+	if xio.Username == "" {
+		return fmt.Errorf("Username cannot be empty")
+	}
+	if xio.Password == "" {
+		return fmt.Errorf("Password cannot be empty")
+	}
+	if xio.URL == "" {
+		return fmt.Errorf("URL cannot be empty")
+	}
+	if len(xio.Collectors) == 0 {
+		xio.Collectors = []string{"bbus","clusters","ssds","volumes","xms"}
+	}
+
+	availableCollectors := []string{"bbus", "clusters", "ssds", "volumes", "xms"}
+	for _, collector := range xio.Collectors {
+		if !choice.Contains(collector,availableCollectors) {
+			return fmt.Errorf("Specified Collector Isn't Supported: " + collector)
+		}
+	}
+	
+	tlsCfg, err := xio.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = tlsCfg
+
+	return nil
+}
+
+func (xio *XtremIO) Gather(acc telegraf.Accumulator) error {
+	var err = xio.authenticate()
+	if err != nil {
+		return err
+	}
+	if xio.Cookie == nil {
+		return fmt.Errorf("no authentication cookie set")
+	}
+
+	var wg sync.WaitGroup
+	for _, collector := range xio.Collectors {
+		wg.Add(1)
+		go func(collector string) {
+			defer wg.Done()
+
+			resp, err := xio.call(collector)
+			if err != nil {
+				acc.AddError(err)
+				return
+			}
+
+			// Due to an inconsistency in the XtremIO API, the XMS endpoint
+			// returns a json array with XMSS as the result. Which is why this
+			// if statement here exists
+			var arr []gjson.Result
+			if collector == "xms" {
+				arr = gjson.Get(resp, "xmss").Array()
+			}else{
```

Comment:
Please care for the formatting and follow other plugins
```suggestion
			} else {
```